### ISSUE TITLE
False positive with `file_header`

### ISSUE THREAD

**Issue by jrpinteno** [realm/SwiftLint#6227]
### New Issue Checklist

- [X] I've Updated SwiftLint to the latest version.
- [X] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

Using `file_header` with a configured pattern triggers violation. This appeared after upgrading to 0.60.0 (it worked fine with 0.59.1 and previous versions).

```swift
// This triggers a violation:

//
//  Test.swift
//  Dummy App
//
//  Created by Xavi R. Pinteño on 3.9.2025.
//  Copyright © 2025 Dummy Corporation. All rights reserved.
//

enum Test {
}
```

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ swiftlint lint Test.swift
```

### Environment

* SwiftLint version 0.60.0
* Xcode version 16.4. Build version 16F6
* Installation method used: Homebrew
* Configuration file:

```yml
opt_in_rules:
  - file_header

file_header:
  required_pattern: |
                    \/\/
                    \/\/  SWIFTLINT_CURRENT_FILENAME
                    \/\/  .*?
                    \/\/
                    \/\/  Created by .*? on \d{1,2}[\.\/]\d{1,2}[\.\/]\d{2,4}\.
                    \/\/  Copyright © \d{4} Dummy Corporation\. All rights reserved\.
                    \/\/
```
* Nested configurations: No

**Comment by neilsardesai** [comment]
+1. Seeing the same thing in 0.60.0

**Comment by lucasecf** [comment]
+1

**Comment by bluerainxty** [comment]
+1

**Comment by pylbert** [comment]
+1

**Comment by egarni** [comment]
+1
